src/lib/message/L1ToL2Message.ts

```diff
@@ -673,6 +677,75 @@ export class EthDepositMessage {
     return ethers.utils.keccak256(rlpEnc)
   }
 
+  /**
+   * Parse the data field in
+   * event InboxMessageDelivered(uint256 indexed messageNum, bytes data);
+   * @param eventData
+   * @returns
+   */
+  private static parseEthDepositData(eventData: string): BigNumber {
+    // https://github.com/OffchainLabs/nitro/blob/9f16d082496aef9de66b9e8653531d467d685560/contracts/src/bridge/Inbox.sol#L230
+    const parsed = ethers.utils.defaultAbiCoder.decode(
+      ['uint256'],
+      eventData
+    ) as [BigNumber]
+
+    return parsed[0]
+  }
+
+  /**
+   * Create an eth deposit messages from a transaction receipt that called ethDeposit
+   * in Inbox.sol
+   * @param l2Provider
+   * @param chainId
+   * @param txReceipt Transaction receipt for a transaction that called ethDeposit
+   * @returns
+   */
+  public static async fromTxReceipt(
+    l2Provider: Provider,
+    txReceipt: L1TransactionReceipt
+  ) {
+    const chainId = (await l2Provider.getNetwork()).chainId
+    return txReceipt
+      .getMessageEvents()
+      .filter(
+        e =>
+          e.bridgeMessageEvent.kind ===
+          InboxMessageKind.L1MessageType_ethDeposit
+      )
+      .map(m => {
+        const value = EthDepositMessage.parseEthDepositData(
+          m.inboxMessageEvent.data
+        )
+
+        // we need to apply an alias to the sender to get the to address, but only if the sender was an EoA
+        // https://github.com/OffchainLabs/nitro/blob/a4eb505ae4f47967fa0ec18e85a43aa4602540c5/contracts/src/bridge/Inbox.sol#L216
+        const aliasedSender = new Address(
+          m.bridgeMessageEvent.sender
+        ).applyAlias().value
+        const to =
+          txReceipt.from.toLowerCase() === aliasedSender.toLowerCase()
+            ? aliasedSender
+            : m.bridgeMessageEvent.sender
```

Comment:
we should always apply the alias here since its the data that comes from the event